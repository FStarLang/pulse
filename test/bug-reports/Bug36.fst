module Bug36

#lang-pulse
open Pulse

[@@expect_failure [189]]
fn foo
  (n : nat) // error shown here: Expected expression of type Type got expression n of type nat
  requires emp
  returns m:nat
  ensures pure (m == n)
{
  id #n // <- but the issue is here
}
