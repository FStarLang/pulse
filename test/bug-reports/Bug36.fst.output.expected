* Info at Bug36.fst(13,6-13,7):
  - Expected failure:
  - Expected expression of type Type got expression n of type Prims.nat

